### PR TITLE
Add a SkinScratchWikiSkin::initPage

### LINKED ISSUE #72: Deprecated: SkinScratchWikiSkin::setupSkinUserCss must not modify the style module list,
In MediaWiki (current master):

`( ! ) Deprecated: SkinScratchWikiSkin::setupSkinUserCss must not modify the style module list, this is deprecated since 1.32 [Called from OutputPage::getRlClient in /vagrant/mediawiki/includes/OutputPage.php at line 2996] in /vagrant/mediawiki/includes/debug/MWDebug.php on line 329`

You can move the body of the `setupSkinUserCss` into `initPage` to safely fix this.

### PR DESCRIPTION
**Pull request by ahmetlii**
Closes #72

### CODE REVIEW

File: ScratchWikiSkin.skin.php [hunk 1]
```diff
@@ -49,17 +49,32 @@ class SkinScratchWikiSkin extends SkinTemplate {
...
+	function setupSkinUserCss( OutputPage $out ) {
 		parent::setupSkinUserCss( $out );
-		$out->addModuleStyles( [
+		$out->addModuleStyles( array(
 			'mediawiki.skinning.interface', 'skins.scratchwikiskin2'
-		] );
-		// make Chrome mobile testing work
-		$out->addMeta('viewport', 'user-scalable=no, initial-scale=1, maximum-scale=1, minimum-scale=1, width=device-width, height=device-height');
+			/* 'skins.foobar' is the name you used in your skin.json file */
+		) );
```
Comment (Kenny2github) [inline ScratchWikiSkin.skin.php:77]: 1. `[]` to `array()` is bad
2. didn't the issue literally say that the fact this call is in this function was the problem? You changed everything *but* what was needed.

**Review by Kenny2github — CHANGES_REQUESTED** [review]

File: ScratchWikiSkin.skin.php (hunk 1 above)
Comment (ahmetlii) [reply ScratchWikiSkin.skin.php]: > 1. `[]` to `array()` is bad
> 2. didn't the issue literally say that the fact this call is in this function was the problem? You changed everything _but_ what was needed.

Sorry, I missed. Fixed the same problem again.

File: ScratchWikiSkin.skin.php [hunk 2]
```diff
@@ -49,15 +49,21 @@ class SkinScratchWikiSkin extends SkinTemplate {
...
+	 * Otherwise you won't need this function and you can safely delete it.
+	 *
+	 * @param OutputPage $out
 	 */
-	public function setupSkinUserCss( OutputPage $out ) {
-		parent::setupSkinUserCss( $out );
+
+	public function initPage( OutputPage $out ) {
+		parent::initPage( $out );
+		$out->addModules('skins.scratchwikiskin2.js');
```
Comment (Kenny2github) [inline ScratchWikiSkin.skin.php:62]: Can you please stop changing things that weren't part of the original issue? There is no such module as `skins.scratchwikiskin2.js`.
```suggestion
```

File: ScratchWikiSkin.skin.php [hunk 3]
```diff
@@ -49,15 +49,21 @@ class SkinScratchWikiSkin extends SkinTemplate {
 		$template = 'ScratchWikiSkinTemplate', $useHeadElement = true;
 
 	/**
-	 * Add CSS via ResourceLoader
+	 * This function adds JavaScript via ResourceLoader
```
Comment (Kenny2github) [inline ScratchWikiSkin.skin.php:52]:
```suggestion
	 * Add CSS via ResourceLoader
```

File: ScratchWikiSkin.skin.php [hunk 4]
```diff
@@ -49,15 +49,21 @@ class SkinScratchWikiSkin extends SkinTemplate {
...
-	public function setupSkinUserCss( OutputPage $out ) {
-		parent::setupSkinUserCss( $out );
+
+	public function initPage( OutputPage $out ) {
+		parent::initPage( $out );
+		$out->addModules('skins.scratchwikiskin2.js');
 		$out->addModuleStyles( [
 			'mediawiki.skinning.interface', 'skins.scratchwikiskin2'
-		] );
+			/* 'skins.foobar' is the name you used in your skin.json file */
```
Comment (Kenny2github) [inline ScratchWikiSkin.skin.php:65]:
```suggestion
```

File: ScratchWikiSkin.skin.php [hunk 5]
```diff
@@ -49,15 +49,21 @@ class SkinScratchWikiSkin extends SkinTemplate {
...
+		 ] );
```
Comment (Kenny2github) [inline ScratchWikiSkin.skin.php:66]: ...why, why is there an indentation change?
```suggestion
		] );
```

**Review by Kenny2github — CHANGES_REQUESTED** [review]

File: ScratchWikiSkin.skin.php [hunk 6]
```diff
@@ -51,10 +51,14 @@ class SkinScratchWikiSkin extends SkinTemplate {
 	/**
 	 * Add CSS via ResourceLoader
 	 *
-	 * @param $out OutputPage
+	 * Use this function if your skin has a JS file(s).
+	 * Otherwise you won't need this function and you can safely delete it.
+	 *
```
Comment (Kenny2github) [inline ScratchWikiSkin.skin.php:56]: no boilerplate detritus please
```suggestion
```

File: ScratchWikiSkin.skin.php [hunk 7]
```diff
@@ -75,4 +79,3 @@ static function onGetPreferences( $user, &$preferences ) {
 		return true;
 	}
 }
-
```
Comment (Kenny2github) [inline ScratchWikiSkin.skin.php:78]: unexplained removal of last line

**Review by Kenny2github — CHANGES_REQUESTED** [review]
mostly done now

File: ScratchWikiSkin.skin.php (hunk 7 above)
Comment (ahmetlii) [reply ScratchWikiSkin.skin.php:78]: that's prob due to my editor (VSCode).

**Review by Kenny2github — APPROVED** [review]

**Pull request merged**